### PR TITLE
Show our data in the gallery

### PR DESCRIPTION
**Pull request by sciunto**
## Description

To pursue on the idea in #3384 I thought that it would be useful to show the data we are shipping with scikit-image. We could separate images for general purposes, scientific images and other specific images (like stereo).

## Checklist
[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests

[For detailed information on these and other aspects see [scikit-image contribution guidelines](http://scikit-image.org/docs/dev/contribute.html)]

 
## For reviewers

(Don't remove the checklist below.)

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: doc/examples/data/plot_general.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+           'page',
+           'text',
+           'rocket',
+           )
+
+#'stereo_motorcycle'
+#'lfw_subset',
+
+fig, axes = plt.subplots(len(images), 1, figsize=(8, 4 * len(images)),
+                         sharex=False, sharey=False)
```
Comment (emmanuelle) [inline doc/examples/data/plot_general.py]: isn't ``sharex/y`` False by default?

File: doc/examples/data/plot_general.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ax = axes.ravel()
+
+for i, image in enumerate(images):
+    caller = getattr(data, image)
+    ax[i].imshow(caller())
+    ax[i].set_title(image)
```
Comment (emmanuelle) [inline doc/examples/data/plot_general.py]: ``ax[i].axis('off')```to gain some space in a crowded figure?

File: doc/examples/data/plot_general.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
+"""
+======================
+General purpose images
```
Comment (emmanuelle) [inline doc/examples/data/plot_general.py]: General-purpose

**Comment by sciunto** [comment]
You are absolutely right for the thumbnail. It's an issue. The idea I got in the mean time to solve this is to take one image as an example for the thumbnail, and then display them all. I'm afraid that, even if we succeed in generating a correct rendering, the future additions will destroy it anyway.

**Comment by emmanuelle** [comment]
@sciunto you can do that by displaying a first image in the example, and then the whole subplots. I don't remember if sphinx-gallery takes the first or the last figure for the thumbnail, and whether it can be configured. The doc of sphinx-gallery is quite thorough, but its organization is sometimes a bit weird.

**Comment by sciunto** [comment]
It's now ready for reviews.

**Comment by sciunto** [comment]
@scikit-image/core Anyone for reviewing this please?

**Comment by sciunto** [comment]
@soupault I increased the font size. Regarding the fact it doesn't show up in section, I guess it's because we use getattr, and not a direct call. A parser cannot get the call here... but i don't think it's significant miss.

File: doc/examples/data/plot_specific.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+===============
+
+"""
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib
+
+from skimage import data
+
+matplotlib.rcParams['font.size'] = 18
```
Comment (soupault) [inline doc/examples/data/plot_specific.py]: I've just noticed that you don't set titles in this example. The font config doesn't really make sense then.

**Comment by soupault** [comment]
> I guess it's because we use getattr, and not a direct call
@sciunto Ah, you are right. That's, most likely, the case.

File: doc/examples/data/plot_specific.py (hunk 4 above)
Comment (soupault) [reply doc/examples/data/plot_specific.py]: Other examples look great!

Comment (sciunto) [reply doc/examples/data/plot_specific.py]: It just to keep the tics homogeneous...

**Comment by soupault** [comment]
Why don't we add the titles here also then? :)

**Comment by sciunto** [comment]
Because we have a header for each image/set of images  that already introduces the content... I can add it, but I did not feel it's necessary.

**Comment by emmanuelle** [comment]
@sciunto I run the example after merging your branch from master, and I got an error because of the ``lbp_frontal_face_cascade_filename`` which was added. A more robust way to exclude datasets could be 
```
exclude = ['data_dir', 'load', 'lbp_frontal_face_cascade_filename']

datasets = data.__all__ # remove load and data_dir
datasets = [dataset for dataset in datasets if not dataset in exclude]

```

**Comment by sciunto** [comment]
@emmanuelle I'm confused. Where do I call `data.__all__`? I couldn't reproduce.

**Comment by emmanuelle** [comment]
@sciunto sorry I got mixed up (with one of my local files). Sorry about that, ignore my comment!

**Comment by emmanuelle** [comment]
LGTM. It could be good to order the sections of the gallery so that the section on data appears on top (as described here https://sphinx-gallery.readthedocs.io/en/stable/advanced_configuration.html?highlight=sorting#sub-gallery-order). But we can also merge this PR first and do the ordering later, depending on your time.

**Comment by sciunto** [comment]
Good point. I open another issue for that. I think that ordering our examples would be useful beyond this PR (eg operation on numpy arrays could come among the first ones).

**Comment by emmanuelle** [comment]
Merging, thanks!

**Pull request merged**